### DISCUSSION TITLE
u8g2 and Heltec WiFi_LoRa32_V2

### DISCUSSION THREAD

**Discussion by nex86** [olikraus/u8g2#1586] [Q&A]
is it supported? 
I have no idea what Constructor to pick, I checked the list and got overwhelmed by the choices quickly..
Anyone here who have this board can help me out?
Help would be much appreciated.

**Comment by olikraus** [comment]
Is there any datasheet of the device?

**Reply by nex86** [reply]
> Is there any datasheet of the device?

I guess?
https://doc.riot-os.org/group__boards__esp32__heltec-lora32-v2.html

"a SSD1306 0.96-inch 128x64 OLED display connected via I2C."

I have tried 
"U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE);"
but it stays black.

**Comment by olikraus** [comment]
I suggest to use U8G2_SSD1306_128X64_NONAME_F_SW_I2C. Probably you need to figure out the pin numbers to which the display is connected. According to this page https://escapequotes.net/esp32-lora-heltec-v2-with-display-pinout-diagram/ pin 15 and pin 4 are used, so maybe
`U8G2_SSD1306_128X64_NONAME_F_SW_I2C u8g2(U8G2_R0, 15, 4, U8X8_PIN_NONE)` might work.
